builder/qemu/config.go

```diff
@@ -65,6 +65,96 @@ type QemuImgArgs struct {
 	Resize  []string `mapstructure:"resize" required:"false"`
 }
 
+// QemuSMPConfig sets the smp configuration option for the Qemu command-line
+//
+// The smp option sets the number of vCPUs to expose to the VM, the final
+// number of available vCPUs is sockets*cores*threads.
+type QemuSMPConfig struct {
+	// The number of virtual cpus to use when building the VM.
+	//
+	// If undefined, the value will either be `1`, or the product of
+	// `sockets * cpus * threads`
+	//
+	// If both any topology element, or this is set, the final value will be the
+	// largest of the two.
```

Comment:
It is not clear to me me which topology elements are being referred to here. Could you reword to call out which elements you are referring to.